Code/Tools/TestImpactFramework/Runtime/Common/Code/Include/Static/BuildTarget/Common/TestImpactBuildGraph.h

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <BuildTarget/Common/TestImpactBuildTargetException.h>
+#include <BuildTarget/Common/TestImpactBuildTargetList.h>
+
+#include <AzCore/std/containers/unordered_map.h>
+#include <AzCore/std/containers/unordered_set.h>
+
+namespace TestImpact
+{
+    template<typename ProductionTarget, typename TestTarget>
+    struct BuildGraphVertex;
+    
+    //! Build graph target set for dependencies or dependers.
+    template<typename ProductionTarget, typename TestTarget>
+    using TargetBuildGraphSet = AZStd::unordered_set<const BuildGraphVertex<ProductionTarget, TestTarget>*>;
+    
+    //! Build graph for the dependencies and dependers of a given build target.
+    template<typename ProductionTarget, typename TestTarget>
+    struct TargetBuildGraph
+    {
+        TargetBuildGraphSet<ProductionTarget, TestTarget> m_build; //!< Build dependencies/dependers.
+        TargetBuildGraphSet<ProductionTarget, TestTarget> m_runtime; //!< Runtime dependencies/dependers.
+    };
+    
+    //! Vertex in the build graph
```

Comment:
suggestion: Any chance of a slightly more detailed description, what actually is the vertex (maybe just clarifying we mean a connection would help?)

Reply:
I've added a bit more context to the comment. Fixed 👍